### ISSUE TITLE
Second argument "_argumentName" prevents guard use in filters

### ISSUE THREAD

**Issue by download13** [rhys-vdw/ts-auto-guard#194]
I was trying to use a type guard in an `array.filter(isType)` expression, but can't seem to because of the second argument on the type guards.

I get this error from typescript:
```
No overload matches this call.
  Overload 1 of 2, '(predicate: (value: any, index: number, array: any[]) => value is SyncCredentials, thisArg?: any): SyncCredentials[]', gave the following error.
    Argument of type '(obj: any, _argumentName?: string | undefined) => obj is SyncCredentials' is not assignable to parameter of type '(value: any, index: number, array: any[]) => value is SyncCredentials'.
      Types of parameters '_argumentName' and 'index' are incompatible.
        Type 'number' is not assignable to type 'string'.
  Overload 2 of 2, '(predicate: (value: any, index: number, array: any[]) => unknown, thisArg?: any): any[]', gave the following error.
    Argument of type '(obj: any, _argumentName?: string | undefined) => obj is SyncCredentials' is not assignable to parameter of type '(value: any, index: number, array: any[]) => unknown'.
      Types of parameters '_argumentName' and 'index' are incompatible.
        Type 'number' is not assignable to type 'string'.ts(2769)
```

presumably because the filter method expects the second parameter on it's predicate (if one exists) to be a number, not a string.

Is there a reason we need the `_argumentName?: string` parameter on the guards? It doesn't appear to be used for anything. Would it be safe to remove?

**Comment by rhys-vdw** [comment]
Hm! Interesting. Hadn't considered this before. The reason it's there is for the [debug mode](https://github.com/rhys-vdw/ts-auto-guard#debug-mode), in which it's essential for making sense of the output.

The only reason I kept the parameter in was because it was slightly more annoying to have it be excluded when debug mode was disabled. But it would be reasonable to change the template.

You're welcome to open a PR that removes it, or you can just do:
```ts
array.filter(e => isFoo(e))
```
I'm sure you're aware of that though.

**Comment by rhys-vdw** [comment]
Closed by #195
